src/test/scala-2.11/markets/engines/BrokenMatchingEngine.scala

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2015 David R. Pugh, J. Doyne Farmer, and Dan F. Tang
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package markets.engines
+
+import markets.orders.{BidOrder, AskOrder, Order}
+
+import scala.collection.{immutable, mutable}
+
+
+/** A BrokenMatchingEngine just stores incoming orders and never generates matches. */
+class BrokenMatchingEngine extends MutableMatchingEngine {
+
+  protected val _askOrderBook = mutable.Set.empty[AskOrder]
+
+  protected val _bidOrderBook = mutable.Set.empty[BidOrder]
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [It's preferred to use immutable collections instead of mutable ones](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1612110834/issues/source?bid=2921362&fileBranchId=3028966#l28)